src/measure_theory/bochner_integration.lean

```diff
@@ -16,92 +45,624 @@ set_option class.instance_max_depth 100
+  by rw finset.sum_add_distrib
+... = ((pair f g).map prod.fst).bintegral + ((pair f g).map prod.snd).bintegral :
+begin
+  rw [map_bintegral (pair f g), map_bintegral (pair f g)],
+  { exact integrable_pair hf hg }, { refl },
+  { exact integrable_pair hf hg }, { refl }
+end
+... = bintegral f + bintegral g : rfl
+
+lemma bintegral_smul (r : ℝ) {f : α →ₛ β} (hf : integrable f) :
+  bintegral (r • f) = r • bintegral f :=
+calc bintegral (r • f) = sum f.range (λx, ennreal.to_real (volume (f ⁻¹' {x})) • r • x) :
+  by rw [smul_eq_map r f, map_bintegral f _ hf (smul_zero _)]
+... = (f.range).sum (λ (x : β), ((ennreal.to_real (volume (f ⁻¹' {x}))) * r) • x) :
+  finset.sum_congr rfl $ λb hb, by apply smul_smul
+... = r • bintegral f :
+begin
+  rw [bintegral, smul_sum],
+  refine finset.sum_congr rfl (λb hb, _),
+  rw [smul_smul, mul_comm]
+end
+
+lemma norm_bintegral_le_bintegral_norm (f : α →ₛ β) (hf : integrable f) :
+  ∥f.bintegral∥ ≤ (f.map norm).bintegral :=
+begin
+  rw map_bintegral f norm hf norm_zero,
+  rw bintegral,
+  calc ∥sum f.range (λx, ennreal.to_real (volume (f ⁻¹' {x})) • x)∥ ≤
+       sum f.range (λx, ∥ennreal.to_real (volume (f ⁻¹' {x})) • x∥) :
+    norm_triangle_sum _ _
+    ... = sum f.range (λx, ennreal.to_real (volume (f ⁻¹' {x})) • ∥x∥) :
+    begin
+      refine finset.sum_congr rfl (λb hb, _),
+      rw [norm_smul, smul_eq_mul, real.norm_eq_abs, abs_of_nonneg to_real_nonneg]
+    end
+end
+
+end bintegral
+
+end simple_func
 
 namespace l1
 
-variables (α γ)
+open ae_eq_fun
+
+variables [normed_group β] [second_countable_topology β]
+          [normed_group γ] [second_countable_topology γ]
+
+variables (α β)
+/-- `l1.simple_func` is a subspace of L1 consisting of equivalence classes of an integrable simple
+    function. -/
 def simple_func : Type* :=
-{ f : α →₁ γ // ∃ (s : simple_func α γ) (hs : integrable s), mk s s.measurable hs = f}
-variables {α γ}
+{ f : α →₁ β // ∃ (s : α →ₛ β),  integrable s ∧ ae_eq_fun.mk s s.measurable = f}
 
-local infixr ` →ₛ `:25 := measure_theory.l1.simple_func
+variables {α β}
+
+infixr ` →₁ₛ `:25 := measure_theory.l1.simple_func
 
 namespace simple_func
-open ae_eq_fun
 
-def mk (f : measure_theory.simple_func α γ) : integrable f → (α →ₛ γ) :=
-assume h,  ⟨l1.mk f f.measurable h , ⟨f, ⟨h, rfl⟩⟩⟩
+section instances
+
+instance : has_coe (α →₁ₛ β) (α →₁ β) := ⟨subtype.val⟩
+
+protected lemma eq {f g : α →₁ₛ β} : (f : α →₁ β) = (g : α →₁ β) → f = g := subtype.eq
+protected lemma eq' {f g : α →₁ₛ β} : (f : α →ₘ β) = (g : α →ₘ β) → f = g := subtype.eq ∘ subtype.eq
+
+@[elim_cast] protected lemma eq_iff {f g : α →₁ₛ β} : (f : α →₁ β) = (g : α →₁ β) ↔ f = g :=
+iff.intro (subtype.eq) (congr_arg coe)
+
+@[elim_cast] protected lemma eq_iff' {f g : α →₁ₛ β} : (f : α →ₘ β) = (g : α →ₘ β) ↔ f = g :=
+iff.intro (simple_func.eq') (congr_arg _)
+
+@[nolint] protected def emetric_space  : emetric_space (α →₁ₛ β) := subtype.emetric_space
```

Comment:
You have put `nolint` attributes on this def and the next one. I would prefer to see a docstring explaining the content and why it is not an instance, on all of them.